Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -2895,7 +2901,378 @@ namespace OpenInfraPlatform
+					const typename IfcEntityTypesT::IfcNonNegativeLengthMeasure& parameter) const noexcept(false)
+				{
+					return getDirectionOfCurve(polynomialCurve, parameter * this->UnitConvert()->getLengthInMeterFactor());
+				}
+				template<>
+				carve::geom::vector<3> getDirectionOfCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcPolynomialCurve>& polynomialCurve,
+					const double& parameter) const noexcept(false)
+				{				
+					//std::vector<typename IfcEntityTypesT::IfcReal> coefficientsX, coefficientsY, coefficientsZ;
+					//std::vector<double> polynomialConstantX, polynomialConstantY, polynomialConstantZ;
+					int polynomialConstantCntX, polynomialConstantCntY, polynomialConstantCntZ;
+					std::vector<double> polynomialConstantX, polynomialConstantY, polynomialConstantZ;
+					double angleX, angleY, angleZ;
+					//get coefficients
+					if (polynomialCurve->CoefficientsX)
+					{
+						//std::vector<double> polynomialConstantX;
+						std::vector<typename IfcEntityTypesT::IfcReal> coefficientsX = polynomialCurve->CoefficientsX;
+						// convert to double
+						polynomialConstantX.resize(coefficientsX.size());
+						std::transform(coefficientsX.begin(), coefficientsX.end(), polynomialConstantX.begin(), [](auto it) { return it; });
+						polynomialConstantCntX = std::size(polynomialConstantX);
+						//calculate angle
+						//angleX = SpiralUtils::AngleByAngleDeviationPolynomial(polynomialConstantX, polynomialConstantCntX, parameter);
+					}
+					if (polynomialCurve->CoefficientsY)
+					{
+						//std::vector<double> polynomialConstantY;
+						std::vector<typename IfcEntityTypesT::IfcReal> coefficientsY = polynomialCurve->CoefficientsY;
+						//convert to double
+						polynomialConstantY.resize(coefficientsY.size());
+						std::transform(coefficientsY.begin(), coefficientsY.end(), polynomialConstantY.begin(), [](auto it) { return it; });
+						polynomialConstantCntY = std::size(polynomialConstantY);
+						//calculate angle
+						//angleY = SpiralUtils::AngleByAngleDeviationPolynomial(polynomialConstantY, polynomialConstantCntY, parameter);
+					}
+					if (polynomialCurve->CoefficientsZ)
+					{
+						//std::vector<double> polynomialConstantZ;
+						std::vector<typename IfcEntityTypesT::IfcReal> coefficientsZ = polynomialCurve->CoefficientsZ;
+						//convert to double
+						polynomialConstantZ.resize(coefficientsZ.size());
+						std::transform(coefficientsZ.begin(), coefficientsZ.end(), polynomialConstantZ.begin(), [](auto it) { return it; });
+						polynomialConstantCntZ = std::size(polynomialConstantZ);
+						//calculate angle
+						//angleZ = SpiralUtils::AngleByAngleDeviationPolynomial(polynomialConstantZ, polynomialConstantCntZ, parameter);
+					}
+					
+					// calculate angle between two polynomial curves
+					//double angle = std::atan2(angleY, angleX);
+
+					if (polynomialConstantCntX>0 && polynomialConstantCntY>0)

```

Comment:
I use polynomialConstantCntXYZ to check if the coefficients exist. 
Because in the beginning, I create polynomialConstantXYZ vectors and then check it in this if statement it should always return true.
If `polynomialConstantXYZ` exist and its size `polynomialConstantCntXYZ `return positive value.